### PR TITLE
102 post confirmation modal

### LINKED ISSUE #102: add popup modal after completion
some cases
- [x] upload post fail (due to connectivity)
- [x] upload post fail (due to other random stuff idk)
- [x] upload post success (make sure it's actually uploaded lol)

### PR DESCRIPTION
**Pull request by owennjpr**
# Description

Added a modal that pops up on the home page to confirm when the user makes a new post

## Issues
Resolve #102 

## Test
- log in
- create a new post
- when you submit your post and come back to the home page the modal should pop up
- create another post
- make sure the modal pops up again

## Possible Downsides

The modal's entering animation sometimes jitters a little bit. The current system for making sure that the modal only appears once is kinda scuffed and requires the home index file to verify expo router segments which isn't amazing.

### CODE REVIEW

File: src/app/(tabs)/home/index.tsx [hunk 1]
```diff
@@ -122,6 +135,29 @@ const Home = () => {
     setRefreshing(false);
   };
 
+  useEffect(() => {
+    if (segments[1] === "post") {
+      setConfirmHasOpened(false);
+    }
+  }, [segments]);
+
+  const checkPostingStatus = useCallback(() => {
```
Comment (LordofMankid) [inline src/app/(tabs)/home/index.tsx:144]: good convention to tie the useCallback to a function callback as that's how they're normally used, makes it a little more reasonable as well.

File: src/app/(tabs)/post/reviewpost.tsx [hunk 2]
```diff
@@ -152,9 +163,8 @@ export default function reviewpost() {
       </KeyboardAvoidingView>
       <RectangleOrangeButton
         text="Post"
-        onPress={() => {
-          router.push("/home");
-          uploadAllImages(postData.imageURIs);
+        onPress={async () => {
```
Comment (LordofMankid) [inline src/app/(tabs)/post/reviewpost.tsx:166]: moved router.push from here. uploadAllImages is an async function, so it passes through and immediately pushes to home. instead, i moved it into the uploadAllImages function and added an await in the mongoDB create function so ensure that we wait for it before moving to the router.push (though you can probably move it back here now that i think about it)

File: api/posts/create.ts [hunk 3]
```diff
@@ -46,6 +46,7 @@ const create = async (props: CreatePostProps) => {
     });
   } catch (error) {
     console.error("Error adding item:", error);
+    throw error;
```
Comment (LordofMankid) [inline api/posts/create.ts:49]: tried with local backend/turning it off, looks it catches the error here and then continues execution, and goes to the success even when the post creation fails. instead we need to propagate the error by throwing the error from the backend so it'll get caught in the try-catch block of uploadAllImages

**Review by LordofMankid — APPROVED** [review]
in the interest of time i fixed added the error catching but look through the comments i made some useful notes abt handling async functions and error propagation

**Review by johnny-t06 — APPROVED** [review]
LGTM, good work!

File: src/app/(tabs)/post/reviewpost.tsx (hunk 2 above)
Comment (owennjpr) [reply src/app/(tabs)/post/reviewpost.tsx:166]: makes sense

File: src/app/(tabs)/home/index.tsx (hunk 1 above)
Comment (owennjpr) [reply src/app/(tabs)/home/index.tsx:144]: got it

File: api/posts/create.ts (hunk 3 above)
Comment (owennjpr) [reply api/posts/create.ts:49]: ohh good catch

**Pull request merged**